master/custom/builders.py

```diff
@@ -331,6 +331,31 @@ def get_builders(settings):
     return all_builders
 
 
+def get_builder_tier(builder: str) -> str:
+    # Strip trailing branch name
+    import re
+    builder = re.sub(r" 3\.[x\d]+$", "", builder)
+
+    for builders, tier in (
+        (STABLE_BUILDERS_TIER_1, TIER_1),
+        (STABLE_BUILDERS_TIER_2,TIER_2),
+        (STABLE_BUILDERS_TIER_3, TIER_3),
+        (STABLE_BUILDERS_NO_TIER, NO_TIER),
+        (UNSTABLE_BUILDERS_TIER_1, TIER_1),
+        (UNSTABLE_BUILDERS_TIER_2, TIER_2),
+        (UNSTABLE_BUILDERS_TIER_3, TIER_3),
+        (UNSTABLE_BUILDERS_NO_TIER, NO_TIER),
```

Comment:
Maybe we should have a data structure holding those? like `STABLE_TIER_LIST = {TIER_1: STABLE_BUILDERS_TIER_1, ...` AND `UNSTABLE_TIER_LIS = {...}`?